### ISSUE TITLE
Video Dimensions Question

### ISSUE THREAD

**Issue by cyodrew** [twilio/video-quickstart-android#555]
> Before filing an issue please check that the issue is not already addressed by the following:
>  * [Video Guides](https://www.twilio.com/docs/api/video)
>  * [Github Issues](https://github.com/twilio/video-quickstart-android/issues)
>  * [Changelog](https://www.twilio.com/docs/api/video/changelogs/android)

> Please ensure that you are not sharing any
[Personally Identifiable Information(PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
or sensitive account information (API keys, credentials, etc.) when reporting an issue.

### Description

I noticed that when setting up a `LocalVideoTrack`, the dimensions don't seem to reflect a vertical orientation. The callback `onFrameDimensionsChanged` in `VideoRenderer.Listener` initially shows that the height is 480 and the width is 640 (640x480). This also holds true when checking the height and width of the `I420Frame` when creating a custom `VideoRenderer`. Only after connecting to the room, the callback will show the correct height of 640 and width of 480 (480x640).

### Steps to Reproduce

1. Add `VideoRenderer.Listener` to a VideoView
2. Begin rendering the view by calling `addRenderer(renderer)` and observe the dimensions in the `onFrameDimensionsChanged` callback
3. Start a room
4. After `onConnected` is called, `onFrameDimensionsChanged` is called again with different dimensions for the frame if orientation differs from the default

#### Code

```kotlin
primaryVideoView.setListener(object : VideoRenderer.Listener {
            override fun onFrameDimensionsChanged(width: Int, height: Int, rotation: Int) {
                Log.d("VideoActivity", "height: $height, width: $width")
            }

            override fun onFirstFrame() {}

        })
```

#### Expected Behavior

Width and height should have values respective to the device's orientation

#### Actual Behavior

Width and height are not changed until a room is connected

#### Reproduces how Often

100%

### Versions

1.0

#### Video Android SDK

5.9.0

#### Android API
29

#### Android Device

Pixel 3a

**Comment by aaalaniz** [comment]
Hey @cyodrew 

What you described is expected behavior. Prior to connecting to a room, the camera capturers will provide frames to sinks based on the resolution from the camera. By default, the cameras provide frames with a rotation value so consumers can render correctly. The device knows to apply the rotation accordingly.

After joining a Group room, the media server will request the publisher to rotate frames because our media server does not support [CVO tags](https://tools.ietf.org/html/rfc7742#section-4). As a result, after connecting to a room you will observe a different resolution because our capturer pipeline has now begun rotating the frames as requested by the server prior to sending over the wire.

Let me know if I can clarify anything. Thanks!

**Comment by cyodrew** [comment]
Hey @aaalaniz,

I appreciate the explanation here. I had a follow up question regarding some of the occurrences that arise once connecting to a room in regards to the callback from `VideoSink` (now that the SDK is at 6.X). I can open up a new issue if necessary.

I noticed that when changing the video format to 1920x1080 (or any higher resolution than standard 640x480), the frames that come in are not correct in width and height. I played around with the room's bandwidth profile by giving it `renderDimensions` of `TrackPriority.HIGH` value to a higher resolution `VideoDimensions.HD_1080P_VIDEO_DIMENSIONS`  and made the local participant's track have a `TrackPriority.HIGH` when publishing it to the room. However, the `videoFrame` that is returned in the `onFrame` callback for the `VideoSink` seems to have a width and height that is much smaller than that despite the broadcast happening on my home WiFi bandwidth. My testing showed it went from 1080x1920 to 528x960 consistently for the duration of the broadcast. Am I misunderstanding these APIs? I can provide some example code for clarification.
